python/pyarrow/parquet/core.py

```diff
@@ -3101,6 +3130,7 @@ def write_table(table, where, row_group_size=None, version='2.6',
                 dictionary_pagesize_limit=None,
                 store_schema=True,
                 write_page_index=False,
+                page_checksum_enabled=False,
                 **kwargs):
     # Implementor's note: when adding keywords here / updating defaults, also
     # update it in write_to_dataset and _dataset_parquet.pyx ParquetFileWriteOptions
```

Comment:
Trying to figure out the dataset version of the change I found this comment. It says the keyword has to be added to `ParquetFileWriteOptions` and `write_to_dataset` also. I have been trying something like this localy:

```python
import pyarrow as pa
import pyarrow.dataset as ds

table_orig = pa.table({'a': [1, 2, 3, 4]})

pformat = pa.dataset.ParquetFileFormat()
# not sure if ParquetFileWriteOptions will be enough here
write_options = pformat.make_write_options(page_checksum_verification=True) 

ds.write_dataset(
    data=table_orig,
    base_dir="sample_dataset",
    format=pformat,
    file_options=write_options,
)

# this should already work
pq_scan_opts = ds.ParquetFragmentScanOptions(
    page_checksum_verification=True
)
pformat = pa.dataset.ParquetFileFormat(default_fragment_scan_options=pq_scan_opts)
dataset = ds.dataset("sample_dataset", format=pformat)
```

Reply:
I added the missing properties and arguments to `ParquetFileWriteOptions`. Additionally, I added another  unit test, `test_page_checksum_verification_write_dataset()`, based on the example you provided.